tests/unit/context.cpp

```diff
@@ -0,0 +1,75 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <../src/context/distributor.hpp>
+
+namespace cocaine {
+namespace {
+
+
+struct engine_mock_t {
+    double u;
+
+    engine_mock_t(double u) : u(u) {}
+
+    auto
+    utilization() const -> double {
+        return u;
+    }
+
+    auto operator->() const -> const engine_mock_t* {
+        return this;
+    }
+
+    auto operator* () const -> const engine_mock_t& {
+        return *this;
+    }
+
+    auto operator* () -> engine_mock_t& {
+        return *this;
+    }
+};
+
+TEST(context, bucket_random_dispatcher) {
```

Comment:
Would be tests for marginal (or out of expected range) cases for init constants and utilization, e.g 0, 1,  1 < some_val or some_val < 0 useful?